examples/Bickley_jet_shallow_water.jl

```diff
@@ -0,0 +1,244 @@
+    gravitational_acceleration=g,
+    coriolis=FPlane(f=f),
+    )
+
+# ## Background state and perturbation
+# 
+# We specify `Ω` to be the background vorticity of the jet in the absence of any perturbations.
+# The initial conditions include a small-ampitude perturbation that decays away from the center
+# of the jet.
+
+  Ω(x, y, z) = 2 * U * sech(y - Ly/2)^2 * tanh(y - Ly/2)
+ uⁱ(x, y, z) =   U * sech(y - Ly/2)^2 + ϵ * exp(- (y - Ly/2)^2 ) * randn()
+ ηⁱ(x, y, z) = -Δη * tanh(y - Ly/2)
+ hⁱ(x, y, z) = model.grid.Lz + ηⁱ(x, y, z)
+uhⁱ(x, y, z) = uⁱ(x, y, z) * hⁱ(x, y, z)
+nothing # hide
+
+# We set the initial conditions for the zonal mass transport `uhⁱ` and height `hⁱ`.
+
+set!(model, uh = uhⁱ , h = hⁱ)
+
+# We compute the total vorticity and the perturbation vorticity.
+
+uh, vh, h = model.solution
+        u = ComputedField(uh / h)
+        v = ComputedField(vh / h)
+  ω_field = ComputedField( ∂x(vh/h) - ∂y(uh/h) )
+   ω_pert = ComputedField( ω_field - Ω )
+
+# ## Running a `Simulation`
+#
+# We pick the time-step that ensures to resolve the surface gravity waves.
+# A time-step wizard can be applied to use an adaptive time step.
+
+simulation = Simulation(model, Δt = 1e-2, stop_time = 150.00)
+
+# ## Prepare output files
+#
+# Define a function to compute the norm of the perturbation on the cross channel velocity.
+# We obtain the `norm` function from `LinearAlgebra`.
+
+using LinearAlgebra, Printf
+
+function perturbation_norm(model)
+    compute!(v)
+    return norm(interior(v))
+end
+nothing # hide
+
+# Choose the two fields to be output to be the total and perturbation vorticity.
+
+outputs = (ω_total = ω_field, ω_pert = ω_pert)
+
+# Build the `output_writer` for the two-dimensional fields to be output.
+# Output every `t = 1.0`.
+
+simulation.output_writers[:fields] =
+    NetCDFOutputWriter(
+        model,
+        (ω = ω_field, ωp = ω_pert),
+        filepath=joinpath(@__DIR__, "Bickley_jet_shallow_water.nc"),
+        schedule=TimeInterval(1.0),
+        mode = "c")
+
+# Build the `output_writer` for the growth rate, which is a scalar field.
+# Output every time step.
+#
+
+simulation.output_writers[:growth] =
+    NetCDFOutputWriter(
+        model,
+        (perturbation_norm = perturbation_norm,),
+        filepath=joinpath(@__DIR__, "perturbation_norm_shallow_water.nc"),
+        schedule=IterationInterval(1),
+        dimensions=(perturbation_norm=(),),
+        mode = "c")
+
+run!(simulation)
+
+# ## Visualize the results
+#
+
+using NCDatasets
+nothing # hide
+
+# This sets the environment to avoid an error in GKSwstype
+
+ENV["GKSwstype"] = "100"
+
+# Define the coordinates for plotting
+
+xf = xnodes(ω_field)
+yf = ynodes(ω_field)
+nothing # hide
+
+# Define keyword arguments for plotting the contours
+
+kwargs = (
+         xlabel = "x",
+         ylabel = "y",
+           fill = true,
+         levels = 20,
+      linewidth = 0,
+          color = :balance,
+       colorbar = true,
+           ylim = (0, Ly),
```

Comment:
If grid changes to `y = (-Ly/2, Ly/2)` then change here is appropriate.

Reply:
Done and thanks for pointing this out!

Reply:
Yes